pkg/controller/component/lifecycle/lfa_common.go

```diff
@@ -0,0 +1,247 @@
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package lifecycle
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	"github.com/apecloud/kubeblocks/pkg/constant"
+	"github.com/apecloud/kubeblocks/pkg/controller/component"
+)
+
+const (
+	podNameList       = "podNameList"
+	podIPList         = "podIPList"
+	podHostNameList   = "podHostNameList"
+	podHostIPList     = "podHostIPList"
+	compName          = "compName"
+	compReplicas      = "compReplicas"
+	deletingCompList  = "deletingCompList"
+	undeletedCompList = "undeletedCompList"
+)
+
+func getClusterPods(ctx context.Context, cli client.Reader, namespace, clusterName string) (map[string]string, error) {
+	// TODO
+	pods, err := component.ListOwnedPods(ctx, cli, namespace, clusterName, "")
+	if err != nil {
+		return nil, err
+	}
+	name := func(pod *corev1.Pod) string {
+		return pod.Name
+	}
+	IP := func(pod *corev1.Pod) string {
+		return pod.Status.PodIP
+	}
+	hostName := func(pod *corev1.Pod) string {
+		return pod.Spec.Hostname
+	}
+	hostIP := func(pod *corev1.Pod) string {
+		return pod.Status.HostIP
+	}
+	return map[string]string{
+		podNameList:     strings.Join(traverse(pods, name), ","),
+		podIPList:       strings.Join(traverse(pods, IP), ","),
+		podHostNameList: strings.Join(traverse(pods, hostName), ","),
+		podHostIPList:   strings.Join(traverse(pods, hostIP), ","),
+	}, nil
+}
+
+func getCompPods(ctx context.Context, cli client.Reader, namespace, clusterName, compName string) (map[string]string, error) {
+	pods, err := component.ListOwnedPods(ctx, cli, namespace, clusterName, compName)
+	if err != nil {
+		return nil, err
+	}
+	name := func(pod *corev1.Pod) string {
+		return pod.Name
+	}
+	IP := func(pod *corev1.Pod) string {
+		return pod.Status.PodIP
+	}
+	hostName := func(pod *corev1.Pod) string {
+		return pod.Spec.Hostname
+	}
+	hostIP := func(pod *corev1.Pod) string {
+		return pod.Status.HostIP
+	}
+	return map[string]string{
+		podNameList:     strings.Join(traverse(pods, name), ","),
+		podIPList:       strings.Join(traverse(pods, IP), ","),
+		podHostNameList: strings.Join(traverse(pods, hostName), ","),
+		podHostIPList:   strings.Join(traverse(pods, hostIP), ","),
+	}, nil
+}
+
+func getComps(ctx context.Context, cli client.Reader, namespace, clusterName string) (map[string]string, error) {
+	cluster := &appsv1alpha1.Cluster{}
+	key := types.NamespacedName{
+		Namespace: namespace,
+		Name:      clusterName,
+	}
+	if err := cli.Get(ctx, key, cluster); err != nil {
+		return nil, err
+	}
+
+	name := func(spec appsv1alpha1.ClusterComponentSpec) string {
+		return spec.Name
+	}
+	replicas := func(spec appsv1alpha1.ClusterComponentSpec) string {
+		return fmt.Sprintf("%d", spec.Replicas)
+	}
+	components := cluster.Status.Components
```

Comment:
The way of retrieving components is incorrect. The components in the cluster.status do not include sharding components, and the information is not real-time.

Reply:
Not used currently, will be fixed in later PR.